tests/kernel/semaphore/semaphore/src/main.c

```diff
@@ -465,96 +591,226 @@ void test_sem_take_timeout_isr(void)
 }
 
 /**
- * @brief Test multiple semaphore take
+ * @brief Test semaphore take operation by multiple threads
+ * @ingroup kernel_semaphore_tests
  * @see k_sem_take()
  */
 void test_sem_take_multiple(void)
 {
 	u32_t signal_count;
 
+	k_sem_reset(&multiple_thread_sem);
+	signal_count = k_sem_count_get(&multiple_thread_sem);
+	zassert_true(signal_count == 0U, "k_sem_reset failed");
+
 	/*
-	 * Signal the semaphore upon which the another thread is waiting.  The
-	 * thread (which is at a lower priority) will cause simple_sem
+	 * Signal the semaphore upon which the another thread is waiting.
+	 * The thread (which is at a lower priority) will cause simple_sem
 	 * to be signalled, thus waking this task.
 	 */
-	k_thread_create(&sem_tid, stack_1, STACK_SIZE,
+
+	k_thread_create(&sem_tid_1, stack_1, STACK_SIZE,
 			sem_take_multiple_low_prio_helper,
 			NULL, NULL, NULL,
 			K_PRIO_PREEMPT(3), K_USER | K_INHERIT_PERMS,
 			K_NO_WAIT);
 
-	k_thread_create(&sem_tid_1, stack_2, STACK_SIZE,
+	k_thread_create(&sem_tid_2, stack_2, STACK_SIZE,
 			sem_take_multiple_mid_prio_helper,
 			NULL, NULL, NULL,
 			K_PRIO_PREEMPT(2), K_USER | K_INHERIT_PERMS,
 			K_NO_WAIT);
 
-	k_thread_create(&sem_tid_2, stack_3, STACK_SIZE,
-			sem_take_multiple_high_prio_helper,
+	k_thread_create(&sem_tid_3, stack_3, STACK_SIZE,
+			sem_take_multiple_high_prio_long_helper,
 			NULL, NULL, NULL,
```

Comment:
Can you explain the logic change here?  You're adding another thread and another semaphore (with "long" in their names), but doing similar operations on them.  The original test was just about selecting the correct (highest priority) thread to wake up, I think, so I don't understand the purpose.  What's the behavior being tested?  And what does "long" mean?

Reply:
sem_tid_3 and sem_tid_4 are the same highest priority, but the waiting time of sem_tid_3 is longer than sem_tid_4.
if some threads are the same priority, the sem given operation should be decided according to waiting time.
so i want to test that if a sem is available, it should be given to the highest priority and longest waiting thread.

Reply:
I meant: can you explain it in the code? :)

Reply:
I will comment in code his idea)